fahrschein/src/main/java/org/zalando/fahrschein/NakadiClient.java

```diff
@@ -54,6 +58,35 @@ public static NakadiClientBuilder builder(URI baseUri) {
         }
     }
 
+    public <T> void publish(String eventName, List<T> events) throws IOException {
+        final URI uri = baseUri.resolve(String.format("/event-types/%s/events", eventName));
+        final ClientHttpRequest request = clientHttpRequestFactory.createRequest(uri, HttpMethod.POST);
+
+        request.getHeaders().setContentType(MediaType.APPLICATION_JSON);
+
+        try (final OutputStream body = request.getBody()) {
+            objectMapper.writeValue(body, events);
+        }
+
+        try (final ClientHttpResponse response = request.execute()) {
+            if ((response.getRawStatusCode() == 207 || response.getRawStatusCode() == 422)) {
+                try (final InputStream is = response.getBody()) {
+                    final BatchItemResponse[] responses = internalObjectMapper.readValue(is, BatchItemResponse[].class);
+                    final List<BatchItemResponse> failed = new ArrayList<>(responses.length);
+                    for (BatchItemResponse batchItemResponse : responses) {
+                        if (!BatchItemResponse.PublishingStatus.SUBMITTED.equals(batchItemResponse.getPublishingStatus())) {
+                            failed.add(batchItemResponse);
+                        }
+                    }
+                    if (!failed.isEmpty()) {
+                        // TODO: attach corresponding events?
```

Comment:
Currently I don't require the events to implement any interface, so there is also no standard way to extract the `eid` for correlation. Could either require an interface or do an instanceof check,